src/io/aleph/dirigiste/LongRingBuffer.java

```diff
@@ -0,0 +1,41 @@
+package io.aleph.dirigiste;
+
+import java.util.Arrays;
+
+public class LongRingBuffer {
+
+    final long[] _values;
+    private final int _size;
+    private int _count;
+    private int _offset;
+
+    public LongRingBuffer(final int size) {
+        _values = new long[size];
+        _size = size;
+    }
+
+    public void add(final long value) {
+        _values[_offset++] = value;
+        _count++;
+        if(_offset>=_size) {
+            _offset = 0;
+        }
+    }
+
+    public long[] toArray() {
+        int cnt = Math.min(_count, _size);
+        long[] vals = Arrays.copyOf(_values, cnt);
+        Arrays.sort(vals);
+        return vals;
+    }
+
+    public double[] toDoubleArray() {
```

Comment:
toSortedDoubleArray

Reply:
Wait, Zach was storing doubles in longs because there's no AtomicDoubleArray in Java. If you're not using AtomicLongArray, why do we need this bit-converting method?

Reply:
https://github.com/arnaudgeiser/dirigiste/pull/2/commits/471eefd7615af87884e458e61383d582f150aea6